sstables/sstables.hh

```diff
@@ -154,6 +154,23 @@ inline sstring state_to_dir(sstable_state state) {
     }
 }
 
+inline sstable_state state_from_dir(sstring dir) {
```

Comment:
s/sstring/std::string_view/

for two reasons:

1. so it's able to handle more types which are convertible to `std::string_view`
2. so we don't need to create a temporary instance of `sstring` and then drop it on the floor, for comparing it with a `const char*`.

Reply:
Good point